client/src/index.ts

```diff
@@ -231,28 +295,51 @@ export class Query<T, R = T> {
     return q;
   }
 
-  // TODO: pagination. Maybe implement different methods for different type of paginations
-  // and one to simply get the first records returned by the query with no pagination.
+  async getPaginated(options?: BulkQueryOptions<T>): Promise<Page<T, R>> {
+    return this.repository.executeQuery(this, options);
+  }
+
   async getMany(options?: BulkQueryOptions<T>): Promise<R[]> {
-    // TODO: use options
-    return this.repository.query(this);
+    const { records } = await this.getPaginated(options);
+    return records;
   }
```

Comment:
Idea:

```ts
async *getMany(size: number, options?: BulkQueryOptions<T>): AsyncGenerator<R[]> {
    let offset = 0;
    let end = false;

    while (!end) {
        const { records } = await this.getPaginated({ ...options, page: { size, offset } });
        yield records;
        offset += size;
        end = !meta.page.more;
    }
}
```

Reply:
@gimenete This is something I came up during the meeting discussion on pagination, might want to clean-up the code, but I'd like your insight on this approach.

Reply:
I like the idea. I'm not sure if `getMany()` is the best method to implement it in.

What if we do this in the Query class?

```ts
async* [Symbol.asyncIterator]() {
    // while ...
    yield this.getMany();
    // ...
  }
```

I'm not an expert on iterators though 😅

Reply:
It actually works!!

Example:

```ts
    for await (const user of client.db.users.filter('full_name', contains('Owner'))) {
      // Do something with each owner
    }
``` 

Code:

```ts
  async *[Symbol.asyncIterator](): AsyncIterableIterator<R> {
    for await (const [record] of this.getIterator(1)) {
      yield record;
    }
  }

  async *getIterator(chunk: number, options: Omit<BulkQueryOptions<T>, 'page'> = {}): AsyncGenerator<R[]> {
    let offset = 0;
    let end = false;

    while (!end) {
      const { records, meta } = await this.getPaginated({ ...options, page: { size: chunk, offset } });
      yield records;

      offset += chunk;
      end = !meta.page.more;
    }
  }
```

PS: I'm no expert either, I only recall three times I've ever needed to implement ``generators``/``iterators``

Reply:
Nice!!